### PR TITLE
feat: skip preferences if not on listing

### PR DESCRIPTION
**Pull request by seanmalbert**
# Pull Request Template

## Issue Overview

This PR addresses #2124

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This updates `retrieveApplicationConfig` to accept a listing and not include preferences if they don't exist on the listing. This also updates fetching listings for the application to only fetch the listing a user is applying to, instead of all of them and then filtering them out.

## How Can This Be Tested/Reviewed?

Apply to a listing that has preferences and you should see preferences in the progress bar as well as in the form. Apply to a listing that has zero preferences and you should no longer see preferences in the progress bar or in the form.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/pages/applications/review/summary.tsx [hunk 1]
```diff
@@ -40,7 +40,12 @@ const ApplicationSummary = () => {
           </h2>
         </div>
 
-        <FormSummaryDetails application={application} listing={listing} editMode />
+        <FormSummaryDetails
+          application={application}
+          listing={listing}
+          hidePreferences={listing.listingPreferences.length === 0}
```
Comment (emilyjablonski) [inline sites/public/pages/applications/review/summary.tsx:46]: Seems like we could do this inside FormSummaryDetails rather than add a prop

**Comment by emilyjablonski** [comment]
Testing locally lgtm, I would love to see a Cypress test in submit-application for this one

**Review by YazeedLoonat — APPROVED** [review]
LGTM

File: sites/public/pages/applications/review/summary.tsx (hunk 1 above)
Comment (seanmalbert) [reply sites/public/pages/applications/review/summary.tsx:46]: There's at least one other case where FormSummaryDetails is called with `hidePreferences` passed in, and in that case, it just passes true to always hide them.

**Comment by jaredcwhite** [comment]
Looks good to me as well :+1:

**Pull request merged**